### DISCUSSION TITLE
Changing font for all ttkbootstrap widgets

### DISCUSSION THREAD

**Discussion by SeewhyO6** [israel-dryer/ttkbootstrap#322] [Q&A] [answered]
Hi everyone

Is there a way to change the font of all ttkbootstrap widgets ?

**Comment by SeewhyO6** [comment, marked as answer]
```
font = "White Rabbit"
style = ttk.Style(self)
style.configure('.', font=(font, 12))
self.option_add('*TCombobox*Listbox.font', (font, 12))
self.option_add('*TCombobox.font', (font, 12))
self.option_add('*TEntry.font', (font, 12))
self.option_add('*TSpinbox.font', (font, 12))
```

**Reply by kgall89** [reply]
Sorry if this is an easy fix but I am having trouble implementing this with a radiobutton, is it possible to change the size of the text/button to all radio buttons in a frame?

**Reply by israel-dryer** [reply]
The font is style specific, so you'd need to create and apply a style to all the radiobuttons inside the frame using the `style` parameter. You can subclass or update an existing style using the `Style.configure` method as @SeewhyO6 has illustrated above.

**Reply by kgall89** [reply]
I apologize for not quite being on that level of understanding yet. I am using a user created theme named "SAS". He is what I have tried:

```
root = ttkb.Window(themename='SAS')
style = ttkb.Style()
style.configure('rb', font=('', 12))
root.option_add('*TRadiobutton.font', ('', 50))

```
```
gender_frame = ttkb.LabelFrame(add_ath_frame, text='Gender', labelanchor='nw', width=755,
                          height=150, style='rb')

```
The error I receive is:

`Traceback (most recent call last):
  ...checkin.py", line 67, in <module>
    style.configure('rb', font=('', 12))
  ...venv/lib/python3.10/site-packages/ttkbootstrap/style.py", line 516, in configure
    ttkstyle = Bootstyle.update_ttk_widget_style(None, style)
  ...venv/lib/python3.10/site-packages/ttkbootstrap/style.py", line 5046, in update_ttk_widget_style
    builder_method = builder.name_to_method(method_name)
  ...venv/lib/python3.10/site-packages/ttkbootstrap/style.py", line 1086, in name_to_method
    func = getattr(StyleBuilderTTK, method_name)
AttributeError: type object 'StyleBuilderTTK' has no attribute ''`

I have also changed the 'Rb' to '.' and this changes all widgets to the larger size, not just the Radiobuttons as expected. Ive also put in Times New Roman for the '' but the error still pops.

Any help in understanding my flaw is much appreciated

**Reply by kgall89** [reply]
Update: After a while of trial and error, I figured it out:


```
style = ttkb.Style()
style.configure('My.TRadiobutton', font=('', 50))
```

```
male_rb = ttkb.Radiobutton(gender_frame,
                    text="Male", variable=gender, value="Male", style='My.TRadiobutton')

```

Sorry for the rookie questions!

**Comment by brobr** [comment]
Hi, I am struggling to use this approach for changing/setting the font in the text field of a ScrolledText widget. This follows "TkDefaultFont". 

Redefining the standard fonts via 
```
self.commonfont = tkfont.nametofont("TkDefaultFont")
self.commonfont.configure(family = "SomeFontOnTheComp", size = 11)
self.textfont = tkfont.nametofont("TkTextFont")
self.textfont.configure(family = "Other font",  size = 14)
```
shows that "TkDefaultFont" is used for text in the ScrolledText field while "TkTextFont" is not.
Thus,  the text in the ScrolledText is always in "SomeFontOnTheComp".
While this works:
```
style = ttk.Style()
style.configure('Outline.Toolbutton', font = ("A third font", 20))
style.configure('TNotebook', tabposition='ne', padding = (4,0,0,0))
```
this does _not_ work:
```
style.configure('TFrame.text', font = ("TkTextFont", 16)) 
# style.configure('TFrame.', font = ("TkTextFont", 16)) 

```
The `self.option_add` approach of the solution does _not_ work for me (what am I doing wrong?):
```
style = ttk.Style()
style.configure('.', font=(self.textfont, 16)) # works for labels, tabs, buttons but does not set the font in the text field, which is still the one defined for "TkDefaultFont",  "SomeFontOnTheComp")
self.option_add('*Outline.Toolbutton.font', ("A third font", 20)) # does not work
self.option_add('*TFrame.text.font', ("TkTextFont")) # does not work
#self.option_add('*TFrame.font', ("TkTextFont")) # does not work
```

What I can do is, after creating a `field=ScrolledText(master)`, to overwrite the style font for this object with a hard-coded font: `field.text.configure(font=('AnotherFontOnTheBox', 16))` or by
`field.text.configure(font=('TkTextFont', 16))` to "Other font". 

So this is doable, but I was wondering how it can be done more centrally, via `style.configure()`, if that is possible.
